membership_extension/models/membership_line.py

```diff
@@ -52,7 +54,7 @@ def _compute_state(self):
             ):
                 line.state = "canceled"
             else:
-                return super(MembershipLine, line)._compute_state()
+                super(MembershipLine, line)._compute_state()
```

Comment:
A more performant way would be to avoid splitting the super line by line. Something like:
```python
no_invoice_lines = self.filtered(lambda line: isinstance(line.id, models.NewId) or not line.account_invoice_id)
cancelled_lines = self.filtered(lambda line: line.account_invoice_id.state == "posted" and line.account_invoice_id.payment_state == "reversed")
for line in no_invoice_lines:
    line.state = line.state or "none"
cancelled_lines.state = "cancelled"
return super(MembershipLine, (self - no_invoice_lines - cancelled_lines)._compute_state()
```